src/com/etendoerp/dependencymanager/datasource/AddDependecyDS.java

```diff
@@ -1,42 +1,363 @@
+    List<Map<String, Object>> result;
     try {
       final String strETDEPPackageVersionId = parameters.get("@ETDEP_Package_Version.id@");
       final PackageVersion packageVersion = OBDal.getInstance().get(PackageVersion.class, strETDEPPackageVersionId);
-      List<PackageDependency> dependencyList = DependencyTreeBuilder.createDependencyTree(packageVersion);
-      for (PackageDependency dependency : dependencyList) {
-        Map<String, Object> map = new HashMap<>();
-        map.put("group", dependency.getGroup());
-        map.put("artifact", dependency.getArtifact());
-        map.put("version", dependency.getVersion());
-        result.add(map);
-      }
+
+      result = getGridData(parameters, packageVersion);
+
     } catch (Exception e) {
       throw new OBException(e.getMessage());
     }
+
+    return result;
+  }
+
+  /**
+   * Retrieves grid data based on the provided parameters and package version.
+   *
+   * @param parameters
+   *     A map containing the parameters used for filtering and sorting the data.
+   * @param packageVersion
+   *     The `PackageVersion` object used to generate the dependency tree.
+   * @return A list of maps where each map represents a row of data with fields for group, artifact, and version.
+   * @throws JSONException
+   *     if there is an error processing the JSON data for filtering and sorting.
+   */
+  private List<Map<String, Object>> getGridData(Map<String, String> parameters,
+      PackageVersion packageVersion) throws JSONException {
+    List<Map<String, Object>> result = new ArrayList<>();
+    List<PackageDependency> dependencyList = DependencyTreeBuilder.createDependencyTree(packageVersion);
+    for (PackageDependency dependency : dependencyList) {
+      Map<String, Object> map = new HashMap<>();
+      map.put(GROUP, dependency.getGroup());
+      map.put(ARTIFACT, dependency.getArtifact());
+      map.put(VERSION, dependency.getVersion());
+
+      result.add(map);
+    }
+    return applyFilterAndSort(parameters, result);
+  }
+
+  /**
+   * Applies filtering and sorting to the given data based on the provided parameters.
+   *
+   * @param parameters
+   *     A map containing the parameters used for filtering and sorting the data.
+   * @param result
+   *     A list of maps representing the data to be filtered and sorted.
+   * @return A list of maps representing the filtered and sorted data.
+   * @throws JSONException
+   *     if there is an error processing the JSON data for filtering and sorting.
+   */
+  private List<Map<String, Object>> applyFilterAndSort(Map<String, String> parameters,
+      List<Map<String, Object>> result) throws JSONException {
+    DependencySelectedFilters selectedFilters = readCriteria(parameters);
+    //artifact Filter
+    if (selectedFilters.getArtifact() != null) {
+      result = result.stream().filter(
+          row -> StringUtils.contains(row.get(ARTIFACT).toString(), selectedFilters.getArtifact())).collect(
```

Comment:
Given that the `ARTIFACT` constant is used both in `PackageUtil` and `UpdateLocalPackagesUtil` classes: isn't it better to have a constants class where it can be placed for both the aforementioned classes to use it, instead of creating the same constant twice?

Reply:
true